lib/browser_crawler/engine.rb

```diff
@@ -97,8 +97,12 @@ def visited_pages
     private
 
     def get_page_links
-      page.all('a').map do |a|
-        a['href']
+      remove_blank_links(page.all('a').map { |a| a['href']})
+    end
+
+    def remove_blank_links(links)
+      links.reject do |link|
+        link.nil? || link == ''
```

Comment:
`link.nil? || link.empty?` is a bit better

Reply:
Good catch!